src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -682,9 +682,14 @@ class TezosNodeOperator(
       val proposalsMap = proposalsState.toMap
       fetchedBlocksData.map {
         case (offset, md) =>
+          val accountIdsWithUpdatedBalances = for {
+            blockHeaderMetadata <- discardGenesis.lift(md.metadata).toList
+            balanceUpdate <- blockHeaderMetadata.balance_updates
+            id <- balanceUpdate.contract.map(_.id).orElse(balanceUpdate.delegate.map(_.value)).toList
```

Comment:
Are we sure that `contract` and `delegate` are mutually exclusive in the update?
Maybe you ask me this already but I didn't get exactly your answer possibly.

We might better find a way to compose those "inclusively", like
`update.contract.map(_.id).toList ::: update.delegate.map(_.value).toList`

Reply:
you are right, I  made this assumption on my own, changed